core/src/core-api/routing.js

```diff
@@ -60,19 +61,44 @@ class LuigiRouting {
     } else {
       this._modifySearchParam(params, url.searchParams);
     }
-    window.history.pushState({}, '', url.href);
+
+    this.handleBrowserHistory(keepBrowserHistory, url.href);
     LuigiConfig.configChanged();
   }
 
-  //Adds and remove properties from searchParams
-  _modifySearchParam(params, searchParams) {
+  // Adds and remove properties from searchParams
+  _modifySearchParam(params, searchParams, paramPrefix) {
     for (const [key, value] of Object.entries(params)) {
-      searchParams.set(key, value);
+      const paramKey = paramPrefix ? `${paramPrefix}${key}` : key;
+
+      searchParams.set(paramKey, value);
       if (value === undefined) {
         searchParams.delete(key);
       }
     }
   }
+
+  addNodeParams(params, keepBrowserHistory) {
```

Comment:
**addNodeParams(params, keepBrowserHistory)**

In other instances you use 

**addNodeParams(~search~Params, keepBrowserHistory)**

It would be great if we removed the 'search' just to be consistent.